jcl/src/jdk.jcmd/share/classes/openj9/tools/attach/diagnostics/tools/Jps.java

```diff
@@ -68,38 +84,56 @@ public static void main(String[] args) {
 		System.exit(rc);
 	}
 	
-	@SuppressWarnings("nls")
 	private static void parseArguments(String[] args) {
 		printApplicationArguments = false;
 		printJvmArguments = false;
 		noPackageName = true;
 		vmidOnly = false;
-		final String HELPTEXT = "jps: Print a list of Java processes and information about them%n"
-				+ "    -J: supply arguments to the Java VM running jps%n"
-				+ "    -l: print the application package name%n"
-				+ "    -q: print only the virtual machine identifiers%n"
-				+ "    -m: print the application arguments%n"
-				+ "    -v: print the Java VM arguments, including those produced automatically%n";
-		for (String a : args) {
-			switch (a) {
-			case "-l":
+
+		for (String arg : args) {
+			if (isInvalidArgument(arg)) {
+				System.out.printf("illegal argument: %s%n%n", arg);
+				interruptWithHelpMessage();
+			}
+			arg = arg.substring(1);
+			for (char singleArg : arg.toCharArray()) {
+				processArgument(singleArg);
+			}
+		}
+	}
+
+	private static boolean isInvalidArgument(String arg) {
```

Comment:
Sorry, we don't want to restrict `q` with other flags though `RI` does so.
Please revert to previous version.

Reply:
Let's stop the churn. It isn't what we discussed before, but the code is already written. I don't think it's a big deal either way. I plan to merge it the way it is once the testing completes.

Reply:
sure, it is fine to me.